### DISCUSSION TITLE
Sharing filters

### DISCUSSION THREAD

**Discussion by Deanobeano2** [cortexrd/Knack-Toolkit-Library#380] [Q&A]
Hi all

I see in Norm's comments on Carl's "KTL Custom User Filters" YouTube tutorial he said "...... you're right that with the basic setup, filters are only stored locally on the user's device and won't sync to other devices or browsers. However, with a bit more configuration, these filters can be shared and synced across all devices via Knack's backend. They can even be shared with other users by making them "public.""

Can this be done, and if so how?

Kind regards 
Dean

**Comment by CSWinnall** [comment]
HI Dean

Yes this can be done but it involves a bit of set up. If you go here:
https://github.com/cortexrd/Knack-Toolkit-Library/wiki/Advanced-Features#public-filters

and follow the setup instructions.  I think you should be able to set up just the user and public filters but you will need to also follow the iFrameWnd setup instructions.

If you need any help with the setup let me know and I'll try and help.

Craig

**Comment by cortexrd** [comment]
Hi Dean,

Craig is right, the documentation contains all you need to setup the Public Filters.  

But I **strongly** recommend that you setup everything.  If you try to extract the bits and parts that you need, changes are high you'll break something.  

I wrote the initial version of the documentation this way, so you could only setup what you needed, but it was so complicated that I gave up and documented everything a s single process.

Imagine this:  I was trying to explain "if you need this but not that, set this up and ignore that.  Then go to step X, or X depending on Y and Z... "

In short, it was a nightmare!!!!

With the default setup, you'll have everything ready.  All features can be disabled individually if you don't need them.  But I'm sure that you'll soon discover how valuable they are, and once you enable them, you'll never look back again.

Norm

**Comment by Deanobeano2** [comment]
Hi Norm

Thank you for pointing out that while this process might require investing more time than I'd initially wanted to, that there were will no doubt be a pay-back down the track because the groundwork has already been done.

Dean
